### ISSUE TITLE
[FEATURE] acc module acc_extra database types other than string

### ISSUE THREAD

**Issue by ar45** [OpenSIPS/opensips#2205]
- Component: `acc` module
  - Type: `acc_extra`

I am using acc_extra vars  to store additional information for CDR records such as pdd, billing etc.
Currently, all those vars are coerced and inserted to the database as string values. 

It would be beneficial if we can configure the type of those vars so we can insert integer or decimal values as well. 

Currently, I have to do extra work using a trigger or external job to transform the data.

**Comment by greenbea** [comment]
I have the same issue with acc_extra saving unset variables as empty strings forcing me to initialize all my acc_extra variables that can possibly be empty to match the data type I use in postgres

**Comment by liviuchircu** [comment]
Yes, the default would be to set the value to `DEFAULT`, rather than `''`.  Let me update the comment-specification...

**Comment by kingsleytart** [comment]
I would like this too. I think setting to DEFAULT is the tidiest solution, and probably the easiest to implement.

**Comment by kingsleytart** [comment]
I would love to say yes, but some people might not have default values defined in some of their DB fields, at which point such a change would make their queries fail, eg:

```
MariaDB [opensips]> INSERT INTO acc (time) VALUES (DEFAULT);
ERROR 1364 (HY000): Field 'time' doesn't have a default value
MariaDB [opensips]>
```

Is it feasible to introduce a module parameter which, if set, enables this behaviour? eg

`modparam("acc", "acc_extra_field_defaults", 1);`

or some other name if you prefer.

**Comment by ar45** [comment]
The reason I thought of using `DEFAULT` was also because of the ability to backport and the simplicity and easy to implement. But we might need a little more than that.

Long term and covering other usecases would be to have flags on a avp or pvar to indicate whether the value is to be quoted and/or escaped and handle NULL automatically. 
For example 
`$avp(name:<q>)` for quoting and SQLEscaping the value all the time. A null value will be an empty string. 
`$avp(name:<Q>)` for quoting and SQLEscaping the value if it is non null.
`$avp(name:<i>)` for quoting a SQL identifier.
`$avp(name:<k>)` for an unquoted string, which can be used as a SQL KEYWORD like `DEFAULT` or `CURRENT_TIMESTAMP`, etc.


Or maybe have the flags as `printf` style formatting `$avp(name:<%s>)` and `$avp(name:<%Q>)`, `$avp(name:<%.2f>)`

**Comment by ar45** [comment]
The above makes me think about how to handle avp array values. Maybe each index should be able to have it's own flags?

**Comment by ar45** [comment]
Related to this #1530
